doc/locale_api.md

```diff
@@ -0,0 +1,97 @@
+> dependencies on YaST or Ruby. It's small enough and could give us a good
+> overview on how much can we save.
+
+Plan:
+1. take the systemd APIs as a sensible starting point.
+2. deviate only where we add value
+
+### Language and Keyboard
+
+#### Systemd
+
+The systemd API for Locale(Language) and Keyboard is this:
+(where the last boolean means Interactive, and the other boolean means Convert)
+
+```
+org.freedesktop.locale1 service
+/org/freedesktop/locale1 object
+NAME                    TYPE      SIG   RESULT/VALUE
+org.freedesktop.locale1 interface -     -
+.SetLocale              method    asb   -
+.SetVConsoleKeyboard    method    ssbb  -
+.SetX11Keyboard         method    ssssbb-
+(all properties are read-only and emit PropertiesChanged)
+.Locale                 property  as    1 "LANG=en_US.UTF-8"
+.VConsoleKeymap         property  s     "cz-lat2-us"
+.VConsoleKeymapToggle   property  s     ""
+.X11Layout              property  s     "cz,us"
+.X11Model               property  s     "pc105"
+.X11Options             property  s     "terminate:ctrl_alt_bksp,grp:shift_togg…
+.X11Variant             property  s     "qwerty,basic"
+```
+
+#### Design of Elementary Layer (lower)
+
+- Just use the systemd API, don't add any API of our own
+
+#### Design of Proposal Layer (uppper)
+
+- when setting the locale, adjust the proposed package selection and keyboard
+  accordingly. And timezone.
+
+Agama.locale1 interface (on Agama/Language1?)
+- is a copy of freedesktop.locale1,
+  forwarding to the elementary layer,
+  and SetLocale does the additional work
+  - Agama...Software...todo(...)
+  - Agama.locale1.SetVConsoleKeyboard(...)
+  - Agama.locale1.SetX11Keyboard(...)
+  - Agama...Timezone...todo(...)
+
+### Timezone
+
+#### Systemd
+
+```
+$ busctl --system introspect org.freedesktop.timedate1 /org/freedesktop/timedate1
+NAME                      TYPE      SIG  RESULT/VALUE     FLAGS
+...
+org.freedesktop.timedate1 interface -    -                -
+.ListTimezones            method    -    as               -
+.SetLocalRTC              method    bbb  -                -
+.SetNTP                   method    bb   -                -
+.SetTime                  method    xbb  -                -
+.SetTimezone              method    sb   -                -
+(properties are read only)
+.CanNTP                   property  b    true             -
+.LocalRTC                 property  b    false            emits-change
+.NTP                      property  b    false            emits-change
+.NTPSynchronized          property  b    false            -
+.RTCTimeUSec              property  t    1681214874000000 -
+.TimeUSec                 property  t    1681214874046139 -
+.Timezone                 property  s    "Europe/Prague"  emits-change
+```
+
+"LocalRTC" means "is the local time zone used for the real time clock",
+so it's !hwclock_in_UTC
+
+#### Design of Elementary Layer (lower)
+
+Just use the systemd API, don't add any API of our own. We will use
```

Comment:
just to make it clear, by using systemd API you mean ahving own service, but use org.freedesktop.timedate1 interface, right?

Reply:
Well, I wanted to omit it completely and use systemd directly, but maybe it is worth it to add a forwarder because it uses a different bus (Agama vs system)

Agama.Locale1 service -> Agama/Locale1 object -> freedesktop.timedate1 interface, forwards to freedesktop/timedate1

Reply:
Well, systemd cannot be used directly as we do not want to modify insts-sys ( or live system ) and only target ( aka installed ) one.

Reply:
Ah, good point :thinking: